src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/StreamHelper.cs

```diff
@@ -419,11 +419,13 @@ internal static string DecodeStream(Stream stream, string characterSet, out Enco
             return content;
         }
 
-        internal static bool TryGetEncoding(string characterSet, out Encoding encoding)
+        internal static bool TryGetEncoding(string? characterSet, out Encoding encoding)
         {
             bool result = false;
             try
             {
+                ArgumentException.ThrowIfNullOrEmpty(characterSet);
```

Comment:
Hmm, I think it is wrong. `Encoding.GetEncoding(null)` returns OS default but Encoding.GetEncoding(string.Empty) throw. We should investigate the last case.

Reply:
I think the desired behaviour is that both cases (`characterSet == null`; `characterSet == string.Empty`) should should return `result = false` and `encoding = UTF8` (the default). Do you agree?

Reply:
The code is in try-catch. So it makes no sense. Please remove.

Reply:
I know it doesn't make sense, I added it to avoid a possible null reference error for characterSet, could point me to a better solution?

Reply:
Null reference error? In TryGetEncoding? It is impossible since `Encoding.GetEncoding(null)` returns OS default, for string.Empty try-catch will catch and set default too. Also it will set default if .Net doesn't know a value in `characterSet`.

Reply:
The warning before the last 2 commits was `Possible null reference argument for parameter 'name' in 'Encoding Encoding.GetEncoding(string name)'.`, and the build fails on warnings

Reply:
Since [System.Text.Encoding]::GetEncoding($null) return OS default without exception I suggest to open new issue in .Net.
Today we could use a workaround with `!` operator and comment with reference to the new .Net issue.

Reply:
I looked a bit more into it
Test:

```ps1
$source = @"
using System;
using System.Text;


    public class EncodingTest
    {
        public static Encoding GetEncoding1(string name)
        {
            return Encoding.GetEncoding(name);
        }

        public static Encoding GetEncoding2(int name)
        {
            return Encoding.GetEncoding(name);
        }
    }
"@

Add-Type -TypeDefinition $source -Language CSharp -ReferencedAssemblies netstandard, System.Text.Encoding

[EncodingTest]::GetEncoding1($null)
#--> ERROR
[EncodingTest]::GetEncoding2($null)
#--> windows-1252

[EncodingTest]::GetEncoding1("")
#--> ERROR
[EncodingTest]::GetEncoding2("")
#--> windows-1252
[EncodingTest]::GetEncoding2(0)
#--> windows-1252
```

In our case we are using `Encoding.GetEncoding(string name)` so it throws an error with null or empty that gets caught, we don't have a bug everything works as planned. Unfortunalely `Encoding.GetEncoding(string name)` with null gives us a warning, as you said we could use `!` operator.

If `Encoding.GetEncoding(string name)` with name == null returned `windows-1252` that wouldn't be a desired behaviour for us in this case, because we want to default to UTF-8